dotexporter.py

```diff
@@ -87,21 +102,102 @@ def do_GET(self):
         chain_getHeader = self.query("chain_getHeader")
         system_health   = self.query("system_health")
         runtime_version = self.query("state_getRuntimeVersion")
+        self.current_best = int(chain_getHeader['number'], 16)
+        best_drift=int(datetime.utcnow().timestamp()- self.previous['best_epoch'])
+
+
+        if self.current_best > self.previous['best']:
+           self.previous['best'] = self.current_best
+
+           if(datetime.utcnow().timestamp() >= self.previous['best_epoch']):
+              self.previous['best_epoch'] = int(datetime.utcnow().timestamp())
+              if DEBUG:
+                print("datetime.utcnow().timestamp()" + str(datetime.utcnow().timestamp()))
+           m.append({
+             'name': 'dot_stall_best',
+             'value': 0,
+              'prop': { 'block': self.current_best, 'drift': best _drift  }
+             })
+        else:
+           if DEBUG:
+             print("self.previous['best_epoch'" + str(self.previous['best_epoch']))
+             print("datetime.utcnow().timestamp()" + str(datetime.utcnow().timestamp()))
+             print("best seconds difference = " + str(int(datetime.utcnow().timestamp())- self.previous['best_epoch']))
+           if((datetime.utcnow().timestamp() - self.previous['best_epoch']) > TIMEOUT):
+             m.append({
+              'name': 'dot_stall_best',
+              'value': 1,
+              'prop': { 'block': self.current_best, 'drift': best_drift  }
+              })
+             if DEBUG:
+               print("stall detected for best block")
+           else:
+             m.append({
+               'name': 'dot_stall_best',
+               'value': 0,
+              'prop': { 'block': self.current_best, 'drift': best_drift  }
+           })
+
 
         try:
           chain_getFinalizedHead   = self.query("chain_getFinalizedHead")
           chain_FinalizedHeadBlock = self.query("chain_getBlock", [chain_getFinalizedHead])
-          m.append({
-            'name': 'dot_chain_block_number',
-            'prop': { 'block': 'finalized' },
-            'value': int(chain_FinalizedHeadBlock['block']['header']['number'], 16)
-          })
+          self.current_finalized = int(chain_FinalizedHeadBlock['block']['header']['number'], 16)
+          final_drift=int(datetime.utcnow().timestamp()- self.previous['finalized_epoch'])
+
+          if self.current_finalized > self.previous['finalized']:
+            self.previous['finalized'] = self.current_finalized
+
+            if(datetime.utcnow().timestamp() >= self.previous['finalized_epoch']):
```

Comment:
same here as above. iiuc the timestamp already has to have increased when blocks change